src/preferences/common/default-preference-server.ts

```diff
@@ -54,5 +53,10 @@ export class DefaultPreferenceServer implements PreferenceServer {
         return Promise.resolve(!!preference ? preference.defaultValue : undefined);
     }
 
-    setClient(client: PreferenceClient | undefined) { /* no-op */ }
+    onReady(): Promise<void> { return Promise.resolve(undefined) };
+
+    isValidValue(preferenceName: string, value: any) {
```

Comment:
I would do validation of values separate.

I've looked it is done in vscode. They use a custom parser that excludes comments: https://github.com/Microsoft/node-jsonc-parser and produces a custom AST that is used for validations.

We can use this parser to strip out [comments](https://github.com/Microsoft/node-jsonc-parser#utilities) and then this [package](https://www.npmjs.com/package/jsonschema)  to valiate instances.